File: cloud/Server/Data/Accounts/Account.cs
// Copyright © Spatial Corporation. All rights reserved.

using Spatial.Persistence;

namespace Spatial.Cloud.Data.Accounts;

/// <summary>
/// ...
/// </summary>
[Collection("accounts")]
public class Account : Record
{
    /// <summary>
    /// The user's name.
    /// </summary>
    public string Name { get; set; }

    /// <summary>
    /// The user's email address.
    /// </summary>
    public string Email { get; set; }

    /// <summary>
    /// The user's avatar.
    /// </summary>
    public string? Avatar { get; set; }
}

File: interface/data/accounts/Account.ts
// Copyright © Spatial Corporation. All rights reserved.

import { Resource } from "..";

export type Account = Resource<{
  /**
   * The user's name.
   */
  name: string;

  /**
   * The user's email address.
   */
  email: string;

  /**
   * The user's avatar.
   */
  avatar?: string;
}>;
